### PR TITLE
Add filesystem detect command

### PR DESCRIPTION
**Pull request by utopiabound**
Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1616)

### CODE REVIEW

File: iml-manager-cli/src/filesystem.rs [hunk 1]
```diff
@@ -49,6 +63,44 @@ fn usage(
...
+        hostlist
+            .iter()
+            .filter_map(|h| hostmap.get(h))
+            .cloned()
+            .collect()
+    } else {
+        vec![]
+    };
+    tracing::debug!("Host APIs: {:?}", hosts);
+    let cmd = SendCmd {
```
Comment (jgrund) [inline iml-manager-cli/src/filesystem.rs]:
```suggestion
    let hosts = if let Some(hl) = hosts {
        let hostlist = hostlist_parser::parse(&hl)?;
        tracing::debug!("Host Names: {:?}", hostlist);
        let all_hosts = get_hosts().await?;

        let hostmap: BTreeMap<&str, &str> = all_hosts
            .objects
            .iter()
            .map(|h| {
                vec![
                    (h.nodename.as_str(), h.resource_uri.as_str()),
                    (h.fqdn.as_str(), h.resource_uri.as_str()),
                ]
            })
            .flatten()
            .collect();

        hostlist
            .iter()
            .filter_map(|h| hostmap.get(h.as_str()))
            .map(|x| x.to_string())
            .collect()
    } else {
        vec![]
    };
```

Comment (jgrund) [reply iml-manager-cli/src/filesystem.rs]: We can avoid cloning all the known hosts by doing the above.

File: iml-manager-cli/src/filesystem.rs [hunk 2]
```diff
@@ -49,6 +63,48 @@ fn usage(
...
+                    (h.fqdn.as_str(), h.resource_uri.as_str()),
+                ]
+            })
+            .flatten()
+            .collect();
+
+        hostlist
+            .iter()
+            .filter_map(|h| hostmap.get(h.as_str()))
+            .map(|x| x.to_string())
```
Comment (johnsonw) [inline iml-manager-cli/src/filesystem.rs]: Can this clippy warning be cleaned up?

Comment (jgrund) [inline iml-manager-cli/src/filesystem.rs]:
```suggestion
            .map(|x| (*x).to_string())
```

Comment (jgrund) [reply iml-manager-cli/src/filesystem.rs]: my fault, I'll fix.

Comment (jgrund) [reply iml-manager-cli/src/filesystem.rs]: Fixed.

**Pull request merged**